### ISSUE TITLE
Serializing a Calendar with @JsonAdapter uses default Calendar format

### ISSUE THREAD

**Issue by outzenj** [google/gson#1933]
I'm using the @JsonAdapter annotation to mark a particular Calendar object to use a specific format, like this:

```Java
import java.io.IOException;
import java.text.DateFormat;
import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.Calendar;

import com.google.gson.Gson;
import com.google.gson.TypeAdapter;
import com.google.gson.annotations.JsonAdapter;
import com.google.gson.stream.JsonReader;
import com.google.gson.stream.JsonWriter;

public class TypeA {

	@JsonAdapter(CalendarAdapter.class)
	public Calendar date;


	public static class CalendarAdapter extends TypeAdapter<Calendar> {

		public static final DateFormat DATE_FORMAT = new SimpleDateFormat("yyyy MM dd HH:mm");

		@Override
		public Calendar read(JsonReader reader) throws IOException {
			String value = reader.nextString();
			try {
				Calendar result = Calendar.getInstance();
				result.setTime(DATE_FORMAT.parse(value));
				return result;
			} catch (ParseException pe) {
				throw new IOException(pe);
			}
		}

		@Override
		public void write(JsonWriter writer, Calendar value) throws IOException {
			writer.value(DATE_FORMAT.format(value.getTime()));
		}

	}

	public static void main(String[] args) {
		Gson gson = new Gson();

		TypeA a = new TypeA();
		a.date = Calendar.getInstance();

		String serialized = gson.toJson(a);
		System.out.println(serialized);

	}

}
```

This produces the following output, which uses the GSON-standard Calendar format instead of the format specified by the TypeAdapter:

`{"date":{"year":2021,"month":7,"dayOfMonth":11,"hourOfDay":16,"minute":15,"second":32}}`

**Comment by Marcono1234** [comment]
Which Gson version are you using?
For version 2.8.7 (currently the latest) I see the expected output `{"date":"2021 08 13 00:09"}`.

Are you using a code minimizer, such as Proguard, which might be removing the `JsonAdapter` annotation?

**Comment by outzenj** [comment]
Ugh, my IDE silently included GSON 2.6.2 alongside 2.8.7. Running manually with 2.8.7 alone works fine. Now I just have to fix my IDE.

Sorry for any confusion or effort expended.
